pallets/ds-maps/src/lib.rs

```diff
@@ -0,0 +1,183 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+use frame_support::{
+    codec::{Decode, Encode},
+    decl_error, decl_event, decl_module, decl_storage, dispatch, ensure,
+    weights::{Weight},
+    Parameter,
+};
+use frame_system::ensure_signed;
+use pallet_ds_accounts as accounts;
+use accounts::REGISTRAR_ROLE;
+
+mod default_weight;
+#[cfg(test)]
+mod mock;
+#[cfg(feature = "payment")]
+mod payment;
+#[cfg(test)]
+mod tests;
+
+//Not sure, are those derives required?
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum ZoneType {
+    /// Forbidden type zone
+    Red,
+    /// Available for safe flights
+    Green,
+    /// Owns zones
+    Parent,
+    // 
+}
+
+impl Default for ZoneType {
+    fn default() -> Self {
+        ZoneType::Green
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, Default, Debug, PartialEq, Eq)]
+pub struct BoxCoordinates<CoordinateSize> {
+    coordinates: [CoordinateSize; 6],
+}
+impl<CoordinateSize> BoxCoordinates<CoordinateSize>{
+    pub fn new(coordinates: [CoordinateSize; 6]) -> Self {
+        BoxCoordinates{coordinates}
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, Default)]
+pub struct Zone<CoordinateSize> {
+    pub coordinates: BoxCoordinates<CoordinateSize>,
+    pub zone_type: ZoneType,
+    pub zone_id: u32,
+}
+
+impl<CoordinateSize> Zone<CoordinateSize>
+{
+    pub fn zone_is(&self, zone: ZoneType) -> bool{
+        !(self.zone_type == zone)
+    }
```

Comment:
Suspicious function. The name is `zone_is`, but it checks that zone is NOT of specified type. A mistake or misleading name?